File: src/main/java/com/dochiri/hexagonal/presentation/common/response/ApiResponse.java
package com.dochiri.hexagonal.presentation.common.response;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.LocalDateTime;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiResponse<T>(
        boolean success,
        T data,
        LocalDateTime timestamp
) {

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(
                true,
                data,
                LocalDateTime.now()
        );
    }

}

File: src/main/kotlin/com/dochiri/hexagonal/presentation/common/response/ApiResponse.kt
package com.dochiri.hexagonal.presentation.common.response

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.LocalDateTime

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ApiResponse<T>(
    val success: Boolean,
    val data: T?,
    val timestamp: LocalDateTime
) {
    companion object {
        fun <T> success(data: T): ApiResponse<T> =
            ApiResponse(
                success = true,
                data = data,
                timestamp = LocalDateTime.now()
            )
    }
}
